src/OpenTelemetry.Api/Context/Propagation/BaggagePropagator.cs

```diff
@@ -143,7 +143,10 @@ internal static bool TryExtractBaggage(string[] baggageCollection, out Dictionar
                         break;
                     }
 
-                    if (NameValueHeaderValue.TryParse(pair, out NameValueHeaderValue baggageItem))
+                    var decodedPair = WebUtility.UrlDecode(pair);
+                    var escapedPair = string.Join("=", decodedPair.Split('=').Select(Uri.EscapeDataString));
```

Comment:
Well, we can check `pair` variable for `%` sign, but it may contain '(' or '!' (any symbol from Symptoms section #1607). If we will not escape them `NameValueHeaderValue.TryParse` will return false and we will lose the baggage. As I can see you are the author of this code. Could you please explain what's wrong will happen if we will remove `NameValueHeaderValue.TryParse` call and just split the `pair` by `=` sign?

Reply:
@kipwoker The `NameValueHeaderValue` stuff came from [Microsoft.AspNet.TelemetryCorrelation](https://github.com/aspnet/Microsoft.AspNet.TelemetryCorrelation/blob/b914ebf89d4db4568577af1db1e270f522b05f92/src/Microsoft.AspNet.TelemetryCorrelation/ActivityExtensions.cs#L115). The idea was it would take care of these encoding/decoding/parsing details for us. If it isn't working well, there's no reason to hold onto it.

Reply:
I wrote some tests. Looks like it works well without `NameValueHeaderValue` even if the user escaped their value instead of encoding. Now it looks much simpler to me, also it should work faster and consume less memory. What do you think?